Update Windows implementation to handle app author properly

mysticmind opened the pull request:
- Update Windows implementation to be inline with the Python implementation around handling app author
- Update unit tests
- Update readme to remove the text around Windows support

mysticmind committed:
Update Windows implementation to handle app author

- Update Windows implementation to be inline with the Python implementation
around handling app author
- Update unit tests
- Update readme to remove the text around Windows support
- Fix package by running npm audit

mysticmind committed:
Revert mocha to prior version

mysticmind committed:
Fix handling optional param to appendNameAuthorVersion

mysticmind committed:
Fix a change added which missed passing appauthor

mysticmind commented:
I had to fix few issues, the CI build is green now!

https://travis-ci.org/github/building5/appdirsjs/builds/707959168

leedm777 commented on index.js:
```diff
@@ -33,13 +31,31 @@ var os = require('os'),
  * @private
  */
-var appendNameVersion = function (dir, appname, version) {
+var appendNameAuthorVersion = function (dir, appname, appauthor, version, includeAppAuthor) {
```
Instead of renaming the function and adding a `includeAppAuthor` param, just add `appendNameAuthorVersion` as a new function and leave the old function (and the macOS implementation) alone.

Since this will change the locations of files, this is going to be a breaking change worthy of a 2.0 release. It'd be nice if there was an option to get back to the old behavior, even if that option was deprecated and removed in a 3.0 release. Any thoughts on that?

mysticmind replied:
I have addressed all your queries/comments. I have left the current Windows implementation as the default so that it does not result in a breaking change for the location of files. Included the functionality of including app author in paths for Windows using an opt-in config `includeAppAuthor`.

mysticmind committed:
Revamped the implementation for handling appauthor for windows

- Restored the existing functionality of Windows paths
- Add an opt-in configuration flag `includeAppAuthor` for including appauthor
- Updated units for Windows

mysticmind committed:
Update Travis  CI build config

- Update Travis CI build to run on linux, osx and windows
- Update unit tests to run based on OS env

mysticmind commented:
@leedm777 Appreciate if could review the PR sooner and let me know if you are happy to go ahead and merge it. Please note that I have addressed all your review comments.
